client/util.go

```diff
@@ -232,3 +232,20 @@ func parseFilters(filters []string) string {
 	}
 	return strings.Join(result, " and ")
 }
+
+// HTTPTransportWrapper implements the http.RoundTripper
+// It wraps a *http.Transport, and it is used to add some pre and postprocessing
+// logic to http requests / responses.
+type HTTPTransportWrapper struct {
```

Comment:
I wonder if we can make this an interface called something like `HTTPTransporter` that embeds http.RoundTripper and extends it to add a `Transport() *http.Transport` function which would be implemented by the wrapper.

Reply:
Of course, that makes it much cleaner.

Reply:
Let me know if this is what you had in mind.